WikiWatershed/model-my-watershed
# Adjust layer control so that it is never obscured.

Previously, the element that contained the layer list had a parent with
a very small size and a very deep placement within the map container
hierarchy. To better size and position the list, we make it a direct
sibling of the map container, which is the controlling factor for the
size of the list.

* Refactor layer control code to create two separate elements, one for
the control button and one with the layer list.
* Adjust style of layer list so that it grows and shrinks along with the
map height, is positioned near the top of the map so it has more
vertical space, and so that any hidden layers are accessible by a
scroll bar.
* Adjust style of draw tools container so that it does not overlap the
layer list on small screens.

**Testing Instructions:**
- Start the app from the home page.
- Open the layer control.
- Navigate through the app, and verify that the layer control list is never hidden and changes size based on the map container size.

Thanks to @jfrankl for help figuring this out.

<img width="252" alt="screen shot 2015-09-29 at 2 58 14 pm" src="https://cloud.githubusercontent.com/assets/1042475/10175005/a8b6ac96-66ba-11e5-81e5-55c9f7a993d4.png">

![screen shot 2015-09-29 at 2 58 31 pm](https://cloud.githubusercontent.com/assets/1042475/10175010/abd21f00-66ba-11e5-8b54-fbf6c6f24ecd.png)

![screen shot 2015-09-29 at 2 58 37 pm](https://cloud.githubusercontent.com/assets/1042475/10175013/af0063c6-66ba-11e5-840d-87ad4a4c0887.png)

Connects to #843
